### DISCUSSION TITLE
Virtual File System & search

### DISCUSSION THREAD

**Discussion by DavyLandman** [microsoft/vscode-discussions#40] [Extension Development QnA] [answered]
Is it possible for search operations to traverse the virtual file systems? They can't be opened as projects/workspaces currently, so it might be related to that?

**Comment by r3m0t** [comment, marked as answer]
Yes using proposed APIs `textSearchProvider` and `fileSearchProvider`

**Reply by digitarald** [reply]
Finalization is happening here: https://github.com/microsoft/vscode/issues/59921

**Reply by DavyLandman** [reply]
@digitarald thanks, are there any indications for which release you are aiming?

**Reply by r3m0t** [reply]
@DavyLandman I just noticed your original question says virtual file systems can't be opened as folders. They can, your extension just needs to call `updateWorkspaceFolders` to add the folder to a workspace - or you can generate a `.code-workspace` file and call `executeCommand` to open it.

**Reply by DavyLandman** [reply]
wait @r3m0t, so there is no UI way yet, but I can actually open a folder into the workspace? that sounds good, I'll try that out! (I should have tried it)

**Reply by DavyLandman** [reply]
Ok, indeed, that works. But search doesn't work (also jump to file doesn't work). So that would still require this proposed API to get finished 🤞🏼
